@here/harp-mapview/lib/text/TextElementsRenderer.ts

```diff
@@ -2294,45 +2292,25 @@ export class TextElementsRenderer {
         return true;
     }
 
-    private checkForSmallLabels(textElement: TextElement): THREE.Vector2[] | undefined {
-        let indexOfFirstVisibleScreenPoint = -1;
+    private checkForSmallLabels(textElement: TextElement, screenPoints: THREE.Vector2[]): boolean {
         // Get the screen points that define the label's segments and create a path with
         // them.
-        const screenPoints: THREE.Vector2[] = [];
-        let minX = Number.MAX_SAFE_INTEGER;
-        let maxX = Number.MIN_SAFE_INTEGER;
-        let minY = Number.MAX_SAFE_INTEGER;
-        let maxY = Number.MIN_SAFE_INTEGER;
+        screenPoints.length = 0;
+        let anyPointVisible = false;
+
         for (const pt of textElement.path!) {
             tempPosition.copy(pt).add(textElement.tileCenter!);
```

Comment:
I wonder if this can be avoided somehow, why can't the textElement path positions be pre-transformed to world, this is done on the cpu anyway, so we don't lose any precision.

Reply:
agreed, there's many places in text rendering where we need to transform to world coordinates, so probably we're doing duplicated work. I'll add it to a the labeling performance ticket i created.